### PR TITLE
Add configuration option for monitoring_proxy_url

### PR DESCRIPTION
**Pull request by itzg**
For CMC-1749

## Notes

* The "http_proxy" support included in `net/http` wasn't appropriate since it is designed for proxying HTTP over HTTP and not tunneling TCP like we need
* There were existing Go libraries that provided a pairing of tunnel client and server, but none that implemented just https://tools.ietf.org/rfc/rfc2817

### CODE REVIEW

**Comment by itzg** [comment]
### Note

[This pre-release](https://github.com/racker/rackspace-monitoring-poller/releases/tag/0.2.31.1) was verified to work on a customer deployment.

File: config/config.go [hunk 1]
```diff
@@ -238,6 +241,10 @@ func (cfg *Config) DefineConfigEntries() []configEntry {
 			Allowed:  ValidSnetRegions,
 		},
 		{
+			Name:     "monitoring_proxy_url",
```
Comment (jjbuchan) [inline config/config.go]: We should have a section in the readme with all the possible config options.  Happy for that to be a separate PR though.

Comment (itzg) [reply config/config.go]: Good point since the poller currently supports a subset of the agent's. That can be pulled out pretty easily and would add value for the current PR.
